### PR TITLE
Allow comma separated list of message fields as one query param 

### PR DESCRIPTION
**Pull request by todvora**
Allowing simple request like /api/search/messages?fields=source,http_method,http_response_code
where the fields are comma separated list in one query parameter. 
/nocl

## Motivation and Context
The scripting API should be asi simple and user-friendly as possible. Comma separated list is simpler to type and use then repeated query parameter.

## How Has This Been Tested?
Added unit and integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java [hunk 1]
```diff
@@ -167,7 +170,13 @@ public TabularResponse executeQuery(@ApiParam(name = "query") @QueryParam("query
                                         @ApiParam(name = "metrics") @QueryParam("metrics") List<String> metrics,
                                         @Context SearchUser searchUser) {
         try {
-            AggregationRequestSpec aggregationRequestSpec = queryParamsToFullRequestSpecificationMapper.simpleQueryParamsToFullRequestSpecification(query, streams, timerangeKeyword, groups, metrics);
+            AggregationRequestSpec aggregationRequestSpec = queryParamsToFullRequestSpecificationMapper.simpleQueryParamsToFullRequestSpecification(
+                    query,
+                    StringUtils.splitByComma(streams),
+                    timerangeKeyword,
+                    groups,
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java:177]: Groups should also be split.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java [hunk 2]
```diff
@@ -167,7 +170,13 @@ public TabularResponse executeQuery(@ApiParam(name = "query") @QueryParam("query
...
+                    splitByComma(metrics)
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java:178]: Please use the same variant throughout the code, either with static import everywhere, or without it.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java (hunk 1 above)
Comment (todvora) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java]: done, thx!

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java (hunk 2 above)
Comment (todvora) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java:178]: Fixed

File: graylog2-server/src/test/java/org/graylog2/shared/utilities/StringUtilsTest.java [hunk 3]
```diff
@@ -31,4 +34,16 @@ public void testHumanReadable() {
         assertThat(StringUtils.humanReadableByteCount(1024L * 1024L * 1024L * 1024L * 5L + 1024L * 1024L * 512L)).isEqualTo("5.0 TiB");
         assertThat(StringUtils.humanReadableByteCount(1024L * 5L + 512L)).isEqualTo("5.5 KiB");
     }
+
+    @Test
+    public void testSplitOnComma() {
```
[the remark this answers is not in the document]
Comment (todvora) [reply graylog2-server/src/test/java/org/graylog2/shared/utilities/StringUtilsTest.java]: Good point, done, thx!

**Review by luk-kaminski — APPROVED** [review]
Looks good, thx!

**Pull request merged**